test/test-tty-duplicate-key.c

```diff
@@ -0,0 +1,313 @@
+/* Copyright libuv project contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <io.h>
+#include <windows.h>
+
+#include <string.h>
+#include <errno.h>
+
+#define ESC "\x1b"
+#define EUR_UTF8 "\xe2\x82\xac"
+#define EUR_UNICODE 0x20AC
+
+
+const char* expect_str = NULL;
+ssize_t expect_nread = 0;
+
+static void dump_str(const char* str, ssize_t len) {
+  ssize_t i;
+  for (i = 0; i < len; i++) {
+    fprintf(stderr, "%#02x ", *(str + i));
+  }
+}
+
+static void print_err_msg(const char* expect, ssize_t expect_len,
+    const char* found, ssize_t found_len) {
+  fprintf(stderr, "expect ");
+  dump_str(expect, expect_len);
+  fprintf(stderr, ", but found ");
+  dump_str(found, found_len);
+  fprintf(stderr, "\n");
+}
+
+static void tty_alloc(uv_handle_t* handle, size_t size, uv_buf_t* buf) {
+  buf->base = malloc(size);
```

Comment:
Maybe assert that `buf->base` is not `NULL`.

Reply:
I fixed it, but as far as I see https://github.com/libuv/libuv/pull/2177#issuecomment-460554963, https://github.com/libuv/libuv/pull/2177#issuecomment-461870631 it was better not to fix it at the moment?

Reply:
I think it's different when the code is being written for the first time versus churning over existing code.